File: python/71_flip_the_array_327A.py
n = int(input().strip())
a = list(map(int, input().split()))
ones = a.count(1)
best = -(10**9)
cur = 0
for x in a:
    v = 1 if x == 0 else -1
    cur = max(v, cur + v)
    best = max(best, cur)
print(ones + best)

File: cpp/71_flip_the_array_327A.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    if (!(cin >> n))
        return 0;
    vector<int> a(n);
    int ones = 0;
    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
        if (a[i] == 1)
            ones++;
    }
    int best = -1000, cur = 0;
    for (int i = 0; i < n; i++)
    {
        int v = a[i] == 0 ? 1 : -1;
        cur = max(v, cur + v);
        best = max(best, cur);
    }
    cout << (ones + best) << "\n";
}
